src/Tools/dotnet-monitor/CollectionRules/Actions/CollectStacksAction.cs

```diff
@@ -40,37 +37,33 @@ public ICollectionRuleAction Create(IProcessInfo processInfo, CollectStacksOptio
     internal sealed class CollectStacksAction :
         CollectionRuleEgressActionBase<CollectStacksOptions>
     {
-        private readonly ProfilerChannel _profilerChannel;
+        private readonly IStacksOperationFactory _operationFactory;
 
         public CollectStacksAction(IServiceProvider serviceProvider, IProcessInfo processInfo, CollectStacksOptions options)
             : base(serviceProvider, processInfo, options)
         {
-            _profilerChannel = serviceProvider.GetRequiredService<ProfilerChannel>();
+            _operationFactory = serviceProvider.GetRequiredService<IStacksOperationFactory>();
         }
 
         protected override EgressOperation CreateArtifactOperation(TaskCompletionSource<object> startCompletionSource, CollectionRuleMetadata collectionRuleMetadata)
         {
-            bool isPlainText = Options.GetFormat() == CallStackFormat.PlainText;
+            KeyValueLogScope scope = Utils.CreateArtifactScope(Utils.ArtifactType_Dump, EndpointInfo);
```

Comment:
I'm assuming `ArtifactType_Dump` is a copy-paste error and should be `ArtifactType_Stacks`?